### DISCUSSION TITLE
so-setup failing running docker_container.running on so-dockerregistry

### DISCUSSION THREAD

**Discussion by steelwing** [Security-Onion-Solutions/securityonion#5757] [Unsupported Versions]
I have just freshly installed SecurityOnion 2.3.70 on a server on a disconnected network.  During so-setup, I specified airgap. 
 Running so-setup iso is failing every time at the step of running docker_container.running on container so-dockerregistry.  It fails because that stage attempts to grab something from GitHub.  On the Internet.  It's a disconnected network, so of course it's gonna fail.  Why does it do this when I just told so-setup to do an airgap install??  It makes no sense for an airgap install to even try to reach the internet!

**Comment by steelwing** [comment]
I tried 2.3.80 and it had the same behavior.  The exact error that appears in the log is:
Error ghcr.io/security-onion-solutions/registry:latest: Error 500 Get https://ghcr.io/v2/: net/http request canceled while waiting for connection
This is always going to fail in my environment.  It's a disconnected network.  This makes no sense to me. I've set up SecurityOnion in other disconnected environments.  I've used 2.3.70 in a disconnected environment successfully.  It's working right now in separate disconnected enclave at my worksite.  Why, on this one server, in this environment, does so-setup keep failing this way?  Can someone please help me?

**Reply by influence2006** [reply]
I have this error as well with 2.3.80
Cant seem to fix anything even after a fresh install.

**Comment by jertel** [comment]
Each Security Onion release is tested a number of times, across several installation modes, in an isolated (airgapped) network without access to the Internet. I've checked the setup logs for some of these tests and do not see any attempts to access GitHub. If you can provide the full setup log it may give us a clue as to why your setups are seeing this behavior. My GitHub profile has a GPG public key if you need to encrypt it to protect sensitive data.

**Reply by steelwing** [reply]
As mentioned above, I've set up SecurityOnion in disconnected environments before and never had this problem.  So I know this isn't supposed to happen.  I'll work on getting you those logs.

**Reply by steelwing** [reply]
sosetup.log and install_summary posted below.
[sosetup.log](https://github.com/Security-Onion-Solutions/securityonion/files/7313140/sosetup.log)
[install_summary.txt](https://github.com/Security-Onion-Solutions/securityonion/files/7313141/install_summary.txt)

**Comment by rwwiv** [comment]
@steelwing the failure you're seeing happens when the registry image (and most likely the actual registry) are not copied over successfully during the initial stage of setup. Have you verified the ISO is good using the gpg signature as outline in [VERIFY_ISO.md](https://github.com/Security-Onion-Solutions/securityonion/blob/master/VERIFY_ISO.md)? As a follow up if you have done so, how are you mounting the ISO to this server? We've seen this failure to copy occur when the ISO is mounted over ipmi (iDRAC, iLO, etc) on a slower network connection.

**Reply by steelwing** [reply]
We downloaded the ISO and verified it with sha256 sums and gpg per your directions. We then burned it to DVD and installed from that. The disc is never prompted for thereafter, though we made sure it was present.

**Reply by rwwiv** [reply]
If you mount that DVD are you able to see the `docker/registry_image.tar` and `docker/registry.tar` files, and do the md5 hashes of those files match what's on the ISO? And just to be sure, you're seeing the "Press [Enter] to reboot!" message at the end of the first stage of setup correct?

**Reply by steelwing** [reply]
Apologies for the delayed response.  The DVD does mount on the system, and the md5 hashes for those files on the DVD matches what's in the ISO.  And yes, at the end of the original installation, I get prompted for the reboot, and before the system reboots, the DVD is ejected.  Is the DVD supposed to be mounted during the run of so-setup?  If so, is there a specific path it's supposed to be mounted to?

**Reply by rwwiv** [reply]
The DVD being ejected is completely normal for the install. The files necessary for setup should all be copied to the system's filesystem during that initial stage of the install, though we're obviously not seeing that happen in your case. This could potentially be an issue with the DVD drive or its cable, is USB an option for you?

**Reply by steelwing** [reply]
Because of the environment I work in, USB flash drives are generally not allowed, but I can ask if there's one I can use.  We've been using a USB-connected DVD drive for the installation.

**Reply by rwwiv** [reply]
Is this the same DVD drive you've used on the other systems that you were able to successfully set up as airgapped?

**Reply by steelwing** [reply]
yes, the very same.

**Reply by rwwiv** [reply]
A few suggestions:
1. Double check that the DVD is being verified after burning.
2. Try a different DVD reader if possible.
3. Try a different USB port on the system if available.

To reiterate what @jertel stated: we are unable to replicate this on any of our test systems, which often indicates a hardware issue either in the install medium or in the system itself.

**Reply by steelwing** [reply]
The DVD was verified, yes.  But it just now occurred to me: the disc I burned to was a BlueRay (only thing I had available at the time).  Would that have mattered?
We have other readers, I'll try burning again with one of those (specifically using a DVD, if possible).
